lib/Conversion/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -116,6 +124,177 @@ struct ConvertRLWEDecrypt : public OpConversionPattern<RLWEDecryptOp> {
+          builder.create<random::InitOp>(index0, builder.getI32IntegerAttr(32));
+
+      // Create a uniform discrete random distribution with generated values of
+      // -1, 0, 1.
+      auto uniformDistributionType = random::DistributionType::get(
+          getContext(), random::Distribution::uniform);
+
+      auto uniformDistribution =
+          builder.create<random::DiscreteUniformDistributionOp>(
+              uniformDistributionType, generateRandom,
+              builder.getI32IntegerAttr(-1), builder.getI32IntegerAttr(2));
+
+      // generate random u polynomial from uniform random ternary distribution
+      auto uTensor =
+          builder.create<random::SampleOp>(tensorParams, uniformDistribution);
+      auto u =
+          builder.create<polynomial::FromTensorOp>(uTensor, inputT.getRing());
+
+      // Create a discrete Gaussian distribution
+      auto discreteGaussianDistributionType = random::DistributionType::get(
+          getContext(), random::Distribution::gaussian);
+
+      auto discreteGaussianDistribution =
+          builder.create<random::DiscreteGaussianDistributionOp>(
+              discreteGaussianDistributionType, generateRandom,
+              builder.getI32IntegerAttr(0),
+              builder.getI32IntegerAttr(
+                  5));  // TODO: Add pass options to configure stdev (which is
+                        // currently hardcoded to 5)
+
+      // generate random e0 polynomial from discrete gaussian distribution
+      auto e0Tensor = builder.create<random::SampleOp>(
+          tensorParams, discreteGaussianDistribution);
+      auto e0 =
+          builder.create<polynomial::FromTensorOp>(e0Tensor, inputT.getRing());
+
+      // generate random e1 polynomial from discrete gaussian distribution
+      auto e1Tensor = builder.create<random::SampleOp>(
+          tensorParams, discreteGaussianDistribution);
+      auto e1 =
+          builder.create<polynomial::FromTensorOp>(e1Tensor, inputT.getRing());
+
+      // Computing ciphertext0 = publicKey0 * u + e0 * constantT + input
+      auto publicKey0U = builder.create<polynomial::MulOp>(publicKey0, u);
+      auto tE0 = builder.create<polynomial::MulScalarOp>(e0, constantT);
+      auto pK0UtE0 = builder.create<polynomial::AddOp>(publicKey0U, tE0);
+      auto ciphertext0 = builder.create<polynomial::AddOp>(pK0UtE0, input);
+
+      // Computing ciphertext1 = publicKey1 * u + e1 * constantT
+      auto publicKey1U = builder.create<polynomial::MulOp>(publicKey1, u);
+      auto tE1 = builder.create<polynomial::MulScalarOp>(e1, constantT);
+      auto ciphertext1 = builder.create<polynomial::AddOp>(publicKey1U, tE1);
+
+      // ciphertext = (ciphertext0, ciphertext1)
+      auto ciphertext = builder.create<tensor::FromElementsOp>(
+          llvm::ArrayRef<Value>({ciphertext0, ciphertext1}));
+      rewriter.replaceOp(op, ciphertext);
+    } else {  // secret key
+      // TODO: Add support for secret key encryption. Left as future work.
```

Comment:
Also add to an issue - this would be a good one to mark as good first issue for contributors